File: study-class-app/src/main/java/com/example/studyclassapp/dto/submission/TestSubmitResponse.java
package com.example.studyclassapp.dto.submission;

import com.example.studyclassapp.dto.test.TestResponse;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class TestSubmitResponse {
    private Long id;
    private LocalDateTime startTime;
    private LocalDateTime submitTime;
    private TestResponse test;
    private List<QuestionSubmitResponse> questionSubmitList;

    private Integer totalMark;

    private boolean isDone;
}

File: studty-class-fe/src/types/submission/TestSubmitResponse.ts
import { TestResponse } from "../test/TestResponse"
import { QuestionSubmitResponse } from "./QuestionSubmitResponse"

export type TestSubmitResponse = {
    id: number
    startTime: string
    submitTime: string
    test: TestResponse
    questionSubmitList: Array<QuestionSubmitResponse>
    totalMark: number
    done: boolean
}